examples/applications/restapi/handler_utils.py

```diff
@@ -310,8 +310,9 @@ def _fetch_transaction_dto(self, account: AbstractAccount, tx_id) -> Optional[Di
         return {"tx_hex": tx.to_hex()}
 
     def _wallet_name_available(self, wallet_name) -> bool:
-        available_wallet_names = [
-            os.path.basename(path) for path in self.app_state.daemon.wallets.keys()]
+        available_wallet_names = set(self.all_wallets)
```

Comment:
I'm not sure what this does.

Reply:
hmm actually I think there's a more elegant way to do this... 

There was a bug where after creating a new wallet, it could not find the wallet but actually what I should do is update the self.all_wallets with a "re-scan" (via `os.listdir`) of the wallet directory.